lib/perl/Genome/Config/AnalysisProject/SubjectMapping/Command/Predict/SomaticValidation.pm

```diff
@@ -0,0 +1,247 @@
+            my ($normal_sample) = $normal_sample_set->members();
+            my %data = (
+                tumor_sample => $normal_sample->$id_method,
+                normal_sample => '',
+                snv_variant_list_id => '',
+                indel_variant_list_id => '',
+                sv_variant_list_id => '',
+                tag => 'germline',
+            );
+            $self->add_subject_mapping(\%data);
+        } elsif ($tumor_sample_set) {
+            for my $tumor_sample ($tumor_sample_set->members) {
+                my %data = (
+                    tumor_sample => $tumor_sample->$id_method,
+                    normal_sample => '',
+                    snv_variant_list_id => '',
+                    indel_variant_list_id => '',
+                    sv_variant_list_id => '',
+                    tag => 'tumor-only',
+                );
+                $self->add_subject_mapping(\%data);
+            }
+        }
+    }
+    my $writer = Genome::Utility::IO::SeparatedValueWriter->create(
+        output => $self->file_path,
+        separator => "\t",
+        headers => \@headers,
+        print_headers => 0,
+    );
+
+    for my $subject_mapping ($self->new_subject_mapping_set->members) {
+        $writer->write_one($subject_mapping);
+    }
+    $writer->output->close();
+
+    $self->status_message('Predicted '. $self->new_subject_mapping_set->size .' new subject mappings.');
+
+    return $self->new_subject_mapping_set->size;
+}
+
+
+sub existing_subject_mapping_set {
+    my $self = shift;
+
+    if ($self->_existing_subject_mapping_set) { return $self->_existing_subject_mapping_set; }
+
+    my $id_method = $self->sample_identifier;
+    my @subject_mappings;
+    for my $subject_mapping ($self->analysis_project->subject_mappings) {
+
+        my %data;
+        for my $subject_bridge ($subject_mapping->subject_bridges) {
+            $data{$subject_bridge->label} = $subject_bridge->subject->$id_method;
+        }
+
+        for my $input ($subject_mapping->inputs) {
+            $data{$input->key} = $input->value;
+        }
+
+        for my $tag ($subject_mapping->tags) {
+            if (defined($data{tag})) {
+                $self->fatal_message('Unable to handle multiple subject mapping tags!');
+            }
+            $data{tag} = $tag;
+        }
+        push @subject_mappings, \%data;
+    }
+
+    my $subject_mapping_set = Set::Scalar->new(@subject_mappings);
+    $self->_existing_subject_mapping_set($subject_mapping_set);
+
+    $self->status_message('Found '. $self->existing_subject_mapping_set->size .' existing subject mappings.');
+
+    return $self->_existing_subject_mapping_set;
+}
+
+sub new_subject_mapping_set {
+    my $self = shift;
+    if ($self->_new_subject_mapping_set) { return $self->_new_subject_mapping_set; }
+    my $new_subject_mapping_set = Set::Scalar->new();
+    $self->_new_subject_mapping_set($new_subject_mapping_set);
+    return $self->_new_subject_mapping_set;
```

Comment:
This could use a property also named `new_subject_mapping_set` and then call `__new_subject_mapping_set` inside of here to get at the built-in accessor.